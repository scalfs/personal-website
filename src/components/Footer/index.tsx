import { Icon } from 'icons'
import { IconName } from 'icons/icon'

const links = [
  {
    name: 'GitHub',
    href: 'https://github.com/scalfs'
  },
  {
    name: 'LinkedIn',
    href: 'https://linkedin.com/in/vitorescalfoni'
  },
  {
    name: 'Mail',
    href: 'mailto:hello@vitorescalfoni.com'
  }
] as { name: IconName; href: string }[]

const Footer = () => (
  <footer className="mt-16 border-t">
    <div className="mx-auto flex h-32 w-full max-w-5xl items-center justify-between px-4 text-base tracking-tight text-slate-500 md:px-12">
      <div>
        <p>
          Â© 2022, <span className="font-medium">Vitor Escalfoni.</span>
        </p>
        <p>Designed &amp; Built with ðŸ’™ </p>
      </div>

      <div className="flex gap-2">
        {links.map(({ name, href }) => (
          <a
            key={name}
            href={href}
            aria-label={`${name} link`}
            className="flex size-10 items-center justify-center rounded-lg border bg-slate-100 p-2 transition-colors duration-250 ease-in-out-cubic hover:border-slate-400 hover:bg-slate-200 focus:outline-none focus:ring-2 focus:ring-blue-700 focus:ring-offset-2 focus:ring-offset-slate-50"
          >
            <Icon name={name} />
          </a>
        ))}
      </div>
    </div>
  </footer>
)

export default Footer
