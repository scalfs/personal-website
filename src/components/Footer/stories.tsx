import { ComponentMeta, ComponentStory } from '@storybook/react'
import Footer from '.'

export default {
  title: 'Footer',
  component: Footer
} as ComponentMeta<typeof Footer>

const Story: ComponentStory<typeof Footer> = () => <Footer />

export const Default = Story.bind({})
