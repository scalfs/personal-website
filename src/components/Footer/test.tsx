import { render, screen } from '@testing-library/react'

import Footer from '.'

describe('<Footer />', () => {
  it('should render the section', () => {
    const { container } = render(<Footer />)

    expect(screen.getByRole('link', { name: /github/i })).toBeInTheDocument()
    expect(screen.getByRole('link', { name: /linkedin/i })).toBeInTheDocument()
    expect(screen.getByRole('link', { name: /mail/i })).toBeInTheDocument()
    expect(screen.getAllByRole('img')).toHaveLength(3)

    expect(container.firstChild).toMatchSnapshot()
  })
})
