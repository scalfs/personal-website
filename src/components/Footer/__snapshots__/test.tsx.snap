// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should render the section 1`] = `
<footer
  class="mt-16 border-t"
>
  <div
    class="mx-auto flex h-32 w-full max-w-5xl items-center justify-between px-4 text-base tracking-tight text-slate-500 md:px-12"
  >
    <div>
      <p>
        Â© 2022, 
        <span
          class="font-medium"
        >
          Vitor Escalfoni.
        </span>
      </p>
      <p>
        Designed & Built with ðŸ’™ 
      </p>
    </div>
    <div
      class="flex gap-2"
    >
      <a
        aria-label="GitHub link"
        class="flex h-10 w-10 items-center justify-center rounded-lg border bg-slate-100 p-2 transition-colors duration-250 ease-in-out-cubic hover:border-slate-400 hover:bg-slate-200 focus:outline-none focus:ring-2 focus:ring-blue-700 focus:ring-offset-2 focus:ring-offset-slate-50"
        href="https://github.com/scalfs"
      >
        <svg
          fill="currentColor"
          height="24"
          role="img"
          viewBox="0 0 496 512"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title>
            LinkedIn
          </title>
          <path
            d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
          />
        </svg>
      </a>
      <a
        aria-label="LinkedIn link"
        class="flex h-10 w-10 items-center justify-center rounded-lg border bg-slate-100 p-2 transition-colors duration-250 ease-in-out-cubic hover:border-slate-400 hover:bg-slate-200 focus:outline-none focus:ring-2 focus:ring-blue-700 focus:ring-offset-2 focus:ring-offset-slate-50"
        href="https://linkedin.com/in/vitorescalfoni"
      >
        <svg
          fill="currentColor"
          height="24"
          role="img"
          viewBox="0 0 1000 1000"
          width="24"
          xml:space="preserve"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title>
            LinkedIn
          </title>
          <path
            d="M313.3 826.9V390.1H173.1v436.8h140.2zM248.4 324c46.7 0 75.8-33.4 75.8-75.3-.8-42.6-29.1-75.3-75.1-75.3-46 0-75.8 32.7-75.8 75.3 0 41.9 29.1 75.3 74 75.3h1.1zM665 380.1c-74.3 0-106.2 41.9-126.6 71v-60.8H398v436.5h140.4V578.2c0-12.8 1-25.5 4.8-34.5 10.2-25.5 33.2-51.8 72-51.8 50.8 0 71.2 39.1 71.2 96.2v238.7h140.4v-255c.1-130.9-69.4-191.7-161.8-191.7z"
          />
          <path
            d="M911.6 10H93.7C49.1 10 10 42.2 10 86.3V906c0 44.4 39.1 84 83.7 84h817.6c44.9 0 78.6-39.8 78.6-84V86.3c.4-44.1-33.6-76.3-78.3-76.3zm-8.1 898.6H99.6c-4.6 0-7.9-3.6-7.9-7.9V96.8c0-2.8 2.6-5.1 5.1-5.1h806.7c2.6 0 5.1 2.6 5.1 5.1v806.7c0 2.3-2.3 5.1-5.1 5.1z"
          />
        </svg>
      </a>
      <a
        aria-label="Mail link"
        class="flex h-10 w-10 items-center justify-center rounded-lg border bg-slate-100 p-2 transition-colors duration-250 ease-in-out-cubic hover:border-slate-400 hover:bg-slate-200 focus:outline-none focus:ring-2 focus:ring-blue-700 focus:ring-offset-2 focus:ring-offset-slate-50"
        href="mailto:hello@vitorescalfoni.com"
      >
        <svg
          fill="none"
          height="24"
          role="img"
          stroke="currentColor"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title>
            Email
          </title>
          <path
            d="M3 8l7.89 5.26a2 2 0 002.22 0L21 8M5 19h14a2 2 0 002-2V7a2 2 0 00-2-2H5a2 2 0 00-2 2v10a2 2 0 002 2z"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
      </a>
    </div>
  </div>
</footer>
`
